packages/web/src/store/playlist-library/helpers.ts

```diff
@@ -409,3 +411,53 @@ export const containsTempPlaylist = (
   }
   return false
 }
+
+/**
+ * Takes a library and returns a list of all temporary playlists from that library
+ * @param library
+ * @returns PlaylistLibraryIdentifier[]
+ */
+export const extractTempPlaylistsFromLibrary = (
+  library: PlaylistLibrary | PlaylistLibraryFolder
+): PlaylistLibraryIdentifier[] => {
+  if (isEmpty(library.contents)) return []
+  return library.contents.reduce((prevResult, nextContent) => {
+    if (nextContent.type === 'folder') {
+      return prevResult.concat(extractTempPlaylistsFromLibrary(nextContent))
+    } else if (nextContent.type === 'temp_playlist') {
+      return prevResult.concat(nextContent)
+    } else {
+      return prevResult
+    }
+  }, [] as PlaylistLibraryIdentifier[])
+}
+
+/**
+ * Takes a library and mapping of temporary playlist ids to their resolved
+ * playlist identifiers, then returns the library (does not mutate original)
+ * with temporary playlists replaced by their resolved playlist identifiers.
+ * @param library
+ * @param tempPlaylistIdToResolvedPlaylist object that maps temporary playlist ids to their resolved playlist identifiers
+ * @returns PlaylistLibrary | PlaylistLibraryFolder
+ */
+export const replaceTempWithResolvedPlaylists = <
+  T extends PlaylistLibrary | PlaylistLibraryFolder
+>(
+  library: T,
+  tempPlaylistIdToResolvedPlaylist: { [key: string]: PlaylistLibraryIdentifier }
```

Comment:
TypeScript recommends using the `Record` type now

```suggestion
  tempPlaylistIdToResolvedPlaylist: Record<string, PlaylistLibraryIdentifier>
```